design/Alameda/design.md

```diff
@@ -0,0 +1,30 @@
+## What is Alameda
+
+Alameda is an intelligent resource orchestrator for Kubernetes, providing the features of autonomous balancing, scaling, and scheduling by using machine learning. Alameda learns the continuing changes of computing and I/O metrics from clusters, predicts the future demands for pods, and intelligently orchestrates underlying resources to fulfill the dynamic resource requests without manual configuration.
+
+For more details, please refer to https://github.com/containers-ai/Alameda
+
+## Features
+
+Alameda provides Rook with the following features:
+
+- Disk health prediction  
+    Based on a disk's S.M.A.R.T. value, Alameda predicts how bad a commodity disk is going to fail in a near future. Rook can stop provisioning volumes from a critical status disk.
+- Performance prediction  
+    Alameda learns patterns from the historical performance metrics of persistent volumes and pools. With this knowledge, Rook can:
+    - have more efficient pod resource configuration planning
+    - have more efficient pod allocation planning
+    - provision volumes from low loading pool
+    - recommend disk replacement time
+- Capacity trending prediction  
+    Alameda can also provide capacity usage trending prediction for Rook's storage providers as well as storages exposed by Rook. Rook knows better **when** to add more disks for more capacity.
+
+## How Alameda works
+
+1. First Alameda data collector gets metrics, events, and logs from Prometheus
```

Comment:
What reports the metrics to prometheus? Is there an Alameda agent that will run on each node to collect them? Or is this something that could be reported by the rook discovery agent?

Reply:
Thanks for your review. Alameda does not use agent. It looks up metrics from Prometheus. To my knowledge, rook agent collects r/w throughput/latency/iops and send them to Prometheus too. (please advise if I am wrong)